File: frontend/src/classes/Movie.ts
export default class Movie
{
    id: string;
    title: string;
    genre: string;
    year: number;
    director: string;
    //rating: number;
    //imdb_tag: string;
    release_date: Date;
    created_at: Date;
    movie_order: number;

    constructor(id: string="", title:string = "", year:number = 0, 
                director:string = "", genre:string = "", release_date:Date, created_at:Date, movie_order:number)
               //rating:number = 0.0, imdb_tag: string = ""
    {
        this.id = id;
        this.title = title;
        this.year = year;
        this.director = director;
        //this.rating = rating;
        //this.imdb_tag = imdb_tag;
        this.genre = genre;
        this.release_date = new Date(release_date);
        this.created_at = created_at;
        this.movie_order = movie_order;
    }
}

File: backend/Movies/Movie.cs
namespace WatchNext.Movies
{
	public class Movie
	{
		required public Guid id{ get; set; }
		required public DateTime created_at { get; set; }
		required public string title { get; set; }
		required public DateTime release_date { get; set; }
		required public string director { get; set; }
		required public string genre { get; set; }
		// TODO: Reintroduce these
		//required public string imdb_id { get; set; }
	}
	// Public facing movie object so users can order their lists
	public sealed class MovieListMovie : Movie
	{
		required public int movie_order { get; set; }
	}
	// Used when reordering movies in the database
	public sealed class MovieListMovieReorder
	{
		required public Guid movie_id { get; set; }
		required public Guid list_id { get; set; }
		required public int movie_order { get; set; }
	}
	public sealed class MovieRegister
	{
		required public string title { get; set; }
		required public DateTime release_date { get; set; }
		required public string director { get; set; }
		required public string genre { get; set; }
	}
	public class UpdateMovieRequest()
	{
		required public Guid id { get; set; }
		public string? title { get; set; }
		public DateTime? release_date { get; set; }
		public string? director { get; set; }
		public string? genre { get; set; }
	}
}
